### PR TITLE
Put pause menu in its own scene

### LINKED ISSUE #1038: Split the pause menu into a separate scene
Right now the pause menu is under the player HUD in the microbe stage scene tree. By splitting it into a separate scene it could be reused in future stage scenes, additionally the UI code would be more organized and the scene tree would be less cluttered.

### PR DESCRIPTION
**Pull request by skyace65**
This makes the pause menu its own scene, and it is instanced in the microbe stage and microbe editor. closes #1038

### CODE REVIEW

File: src/microbe_stage/MicrobeHUD.cs [hunk 1]
```diff
@@ -497,9 +497,4 @@ private void EditorButtonPressed()
     /// <summary>
     ///   Receiver for exiting game from microbe stage.
     /// </summary>
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeHUD.cs:499]: Shouldn't this doc comment be removed as well?

**Comment by hhyyrylainen** [comment]
ok to test

File: scripts/gui/PauseMenu.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System;
+using Godot;
+
+public class PauseMenu : Control
+{
+    public void ResumePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        GetTree().Paused = false;
```
Comment (Kasterisk) [inline scripts/gui/PauseMenu.cs:9]: This causes conflict with the pause button (the small circular one next to the menu button), so if I click resume on the menu while the pause button is still on paused mode, the game unpauses but the button is still active (play icon). Now I'm wondering if the pause button should be removed. @hhyyrylainen Should it just be removed? It's kind of redundant if the player could just pause the game by opening up the pause menu.

Comment (hhyyrylainen) [reply scripts/gui/PauseMenu.cs:9]: I've seen let's plays where players press it. Also it is kinda needed for: https://github.com/Revolutionary-Games/Thrive/issues/192

Comment (Kasterisk) [reply scripts/gui/PauseMenu.cs:9]: Ah, alright then.

Comment (hhyyrylainen) [reply scripts/gui/PauseMenu.cs:9]: So instead they need to use some method to collaborate on when the game should be paused. Perhaps a singleton PauseManager or something could be created to handle that.

Comment (Kasterisk) [reply scripts/gui/PauseMenu.cs:9]: Yeah, that's that's what I thought.

**Comment by skyace65** [comment]
I removed the documentation that doesn't need to be there. Are there any other changes I need to make? Also I'm not good enough at programming to put the pause functionality in a singleton for this.

**Pull request closed without merging**